smac/intensification/successive_halving.py

```diff
@@ -27,26 +27,31 @@ class SuccessiveHalving(Intensifier):
     Supplementary reference: http://proceedings.mlr.press/v80/falkner18a/falkner18a-supp.pdf
 
-    Successive Halving intensifier can operate on two kinds of budgets:
+    Successive Halving intensifier (and Hyperband) can operate on two kinds of budgets:
     1. **'Instances' as budget**:
-        When multiple instances are provided or when run objective is "runtime", this is the criterion used as budgets
-        for successive halving iterations i.e., budget determines how many instances the challengers are evaluated on
-        at a time. Top challengers for the next iteration are selected based on the combined performance across
+        When multiple instances are provided or when run objective is "runtime", this is the criterion used as budget
+        for successive halving iterations i.e., the budget determines how many instances the challengers are evaluated
+        on at a time. Top challengers for the next iteration are selected based on the combined performance across
         all instances used.
 
         If `initial_budget` and `max_budget` are not provided, then they are set to 1 and total number
         of available instances respectively by default.
+
     2. **'Runtime' as budget**:
-        Runtime/Cutoff is used when there is only one instance provided and when run objective is "quality",
+        Runtime is used when there is only one instance provided and when run objective is "quality",
         i.e., budget determines how long the challenger is allowed to run.
-        This is interpreted as wallclock time by SMAc's internal resource limiter by default.
-        But the cutoff can also be passed to the target algorithm as an argument to make it an arbitrary runtime limit,
-        (eg: number of epochs for training a neural network) as long as you disable SMAC's internal resource limiter
-        by setting `limit_resources=False` in scenario.
+        The budget is passed to the target algorithm as an argument to make it an arbitrary runtime limit,
+        (eg: number of epochs for training a neural network).
+        Since this deals with runtime, SMAC's internal resource limiter has to be disabled by setting
+        `limit_resources=False`.
+
+        `initial_budget` and `max_budget` are required parameters for this type of budget.
+        `cutoff` should NOT be provided.
 
-        `initial_budget` is a required parameter for this type of budget.
-        If `max_budget` is not provided, then `cutoff` is used as the default maximum budget.
-        If both are provided, `cutoff` is ignored.
+    Examples for successive halving (and hyperband) can be found here:
+    * Runtime objective and multiple instances *(instances as budget)*: `examples/spear_qcp/SMAC4AC_spear_qcp.py`
+    * Quality objective and multiple instances *(instances as budget)*: `examples/SMAC4HPO_svm_hyperband_instances.py`
+    * Quality objective and single instance *(runtime as budget)*: `examples/SMAC4HPO_mlp_hyperband.py`
```

Comment:
I still find this confusing as the budget is not really the runtime, but a positive, real-valued number. Can we rename this to something which reflects this?

Reply:
Can we call it "arbitrary" budget? Since it can represent anything the user wants it to in the TA.

Reply:
How about calling them real-valued? "arbitrary" has a negative connotation for me.